### PR TITLE
feat: FeatureTypes crud - without update (MAPCO-4721)

### PR DESCRIPTION
**Pull request by RonitKissis**
FeatureTypes crud without update ability

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: openapi3.yaml
Comment (CL-SHLOMIKONCHA) [inline openapi3.yaml:1]: featuresType -> featureType

File: openapi3.yaml [hunk 1]
```diff
@@ -413,6 +413,192 @@ paths:
...
+        - in: path
+          name: dataStore
+          schema:
+            type: string
+          required: true
+        - in: query
+          name: list
+          schema:
+            type: string
+            enum: [all, configured, available, available_with_geom]
```
Comment (CL-SHLOMIKONCHA) [inline openapi3.yaml:437]: `available_with_geom` check the response , and if its unnecessary, remove.

File: src/common/geoserver/models/featureType.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+export interface GeoserverPostAttribute extends GeoserverAttribute {
+  source: string;
+}
+
+export interface GeoServerCreateFeatureRequest {
+  featureType: {
+    name: string;
+    nativeName: string;
+    srs: string;
+    attributes: { attribute: GeoserverPostAttribute[] };
```
Comment (CL-SHLOMIKONCHA) [inline src/common/geoserver/models/featureType.ts:93]: GeoserverAttributes

File: src/featureTypes/models/featureTypesManager.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      });
+      throw new ConflictError(errorMessage);
+    }
+    await this.geoserverManager.postRequest<GeoServerCreateFeatureRequest>(
+      `workspaces/${workspaceName}/datastores/${dataStoreName}/featuretypes`,
+      createFeatureRequestBody
+    );
+  }
+
+  private async checkFeature(workspaceName: string, dataStoreName: string, name: string, list: string): Promise<boolean> {
```
Comment (CL-SHLOMIKONCHA) [inline src/featureTypes/models/featureTypesManager.ts:111]: use the enum

File: tests/mocks/featureTypesMocks.ts [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+    maxFeatures: 0,
+    numDecimals: 0,
+    padWithZeros: false,
+    forcedDecimal: false,
+    overridingServiceSRS: false,
+    skipNumberMatched: false,
+    circularArcPresent: false,
+    attributes: {
+      attribute: [
+        {
```
Comment (CL-SHLOMIKONCHA) [inline tests/mocks/featureTypesMocks.ts:93]: check for required fields

File: src/utils/featureConstants.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+export const boundingBox = {
+  minx: -180,
+  maxx: 180,
+  miny: -90,
+  maxy: 90,
+  crs: srs,
+};
+
+export const attributesMapping = [
```
Comment (CL-SHLOMIKONCHA) [inline src/utils/featureConstants.ts:13]: check for required fields

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: src/common/geoserver/models/featureType.ts (hunk 2 above)
Comment (RonitKissis) [reply src/common/geoserver/models/featureType.ts:93]: GeoserverPostAttribute extends GeoserverAttributes with added source attribute for the mapping

File: tests/mocks/featureTypesMocks.ts (hunk 4 above)
Comment (RonitKissis) [reply tests/mocks/featureTypesMocks.ts:93]: nullable are only: source_id, description. cities, countries

File: src/utils/featureConstants.ts (hunk 5 above)
Comment (RonitKissis) [reply src/utils/featureConstants.ts:13]: nullable are only: source_id, description. cities, countries

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

**Pull request merged**